base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java

```diff
@@ -187,6 +206,101 @@ private void scheduleNow() {
         scheduling.set(false);
     }
 
+    private CompletableFuture<Boolean> runCommand(BalanceCommand command) {
+        if (command.getExpectedVer() != null) {
+            Long prevCMDVer = historyCommandCache.getIfPresent(command.getKvRangeId());
+            if (prevCMDVer != null && prevCMDVer >= command.getExpectedVer()) {
+                logWarn("[{}]Command version is duplicated with prev one: {}", storeClient.clusterId(), command);
```

Comment:
clusterId prefix is redundant, log file name contains the information

Reply:
fixed in next commit